resqpy/well/_trajectory.py

```diff
@@ -231,7 +229,10 @@ def __choose_init_method(self, set_tangent_vectors, data_frame, cell_kji0_list,
         elif wellspec_file:
             return 'wellspec_file'
         elif deviation_survey_file:
-            return 'deviation_survey_file'
+            return 'ascii_trajectory_file'
+        elif set_tangent_vectors and type(self.knot_count) is int and self.tangent_vectors is None:
```

Comment:
Hmm, line 162 basically means that this set tangents logic will not be applied when loading from uuid (existing RESQML object).  As that was the whole point of having 'tangent_vectors' as an init option, we need to change something.  Today my feeling is strip that option out of the choose_init_method and init_function_dict after all.  We will treat the setting of tangents as an optional activity after whatever init method (including loading from uuid) has been applied.  In other words, let's put in a function call at lines 163 and 218 with that function applying the test that is currently here (line 233) and then calling self.set_tangents() if required.  Sorry to mess you about with a different steer every day!

Reply:
Not at all! This is why I've learnt more about resqpy in the last couple weeks than all the previous months I've been on the team.